session.go

```diff
@@ -735,18 +744,29 @@ func (q Query) String() string {
 
 //Attempts returns the number of times the query was executed.
 func (q *Query) Attempts() int {
-	attempts := 0
-	for _, metric := range q.metrics {
+	q.metrics.l.Lock()
+	defer q.metrics.l.Unlock()
```

Comment:
I'd handle this the same way with the unlock as you did in the other functions below – and not take the defer hit and unlock after the for loop. There's no other code path executed so the defer isn't necessary.